### PR TITLE
Python 3.11+ support

### PR DESCRIPTION
**Pull request by vEpiphyte**
* Prevent importing synapse package on interpreters < 3.11
* Replace setup.py + MANIFEST.in with pyproject.toml
* Remove unused appveyor configuration
* Remove python version specific axon test behavior
* Remove some older 3.7 related language from dev facing documentation

### CODE REVIEW

File: docker/images/jsonstor/Dockerfile [hunk 1]
```diff
@@ -2,8 +2,7 @@
 FROM vertexproject/vtx-base-image:py311
 
 COPY synapse /build/synapse/synapse
-COPY setup.py /build/synapse/setup.py
-COPY MANIFEST.in /build/synapse/MANIFEST.in
+COPY pyproject.toml /build/synapse/pyproject.toml
```
Comment (mikemoritz) [inline docker/images/jsonstor/Dockerfile:5]: Any specific reason this didn't get the README copied in like the others?

**Review by blackout-vtx — APPROVED** [review]

**Comment by blackout-vtx** [comment]
It looks like the `python37` CI is still required but I'm not sure why. I see you removed it from the CircleCI config. Maybe github has something cached? Or it's in GitHub settings that I don't have access to?

File: .readthedocs.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ version: 2
 build:
   os: ubuntu-22.04
   tools:
-    python: "3.10"
+    python: "3.11"
```
Comment (mikemoritz) [inline .readthedocs.yaml:6]: grepping the docs dir I see some references to python 3.7; might be good to clean those up / tweak here.

File: docker/images/jsonstor/Dockerfile (hunk 1 above)
Comment (vEpiphyte) [reply docker/images/jsonstor/Dockerfile:5]: That is a mistake.

File: .readthedocs.yaml (hunk 2 above)
Comment (vEpiphyte) [reply .readthedocs.yaml:6]: Will go and clean those up.

**Comment by vEpiphyte** [comment]
@blackout-vtx that is a branch protection configuration for github. It would be altered after this is merged.

**Review by Cisphyx — APPROVED** [review]

**Review by mikemoritz — APPROVED** [review]

**Review by blackout-vtx — APPROVED** [review]

**Review by invisig0th — APPROVED** [review]

**Pull request merged**